presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -878,6 +878,12 @@ public void testInPredicateWithCasts()
                         cast(C_SMALLINT, BIGINT),
                         new InListExpression(ImmutableList.of(toExpression(1L, BIGINT)))),
                 withColumnDomains(ImmutableMap.of(C_SMALLINT, Domain.singleValue(SMALLINT, 1L))));
+
+        ExtractionResult result = fromPredicate(new InPredicate(
+                cast(C_BIGINT, INTEGER),
+                new InListExpression(ImmutableList.of(toExpression(1L, INTEGER)))));
+        assertEquals(result.getRemainingExpression(), equal(cast(C_BIGINT, INTEGER), toExpression(1L, INTEGER)));
+        assertEquals(result.getTupleDomain(), TupleDomain.all());
```

Comment:
You could have comment explaining that the expected is not optimal.

Also, a test case with `cast DECIMAL(10,5) → DECIMAL(7,2)` would be interesting, since many original values map to single target value.

Reply:
I played with it a bit, but `LiteralInterpreter` does work with decimals, so I am going to leave it as it is for now.
https://gist.github.com/219afbdc5a0315aa394a731b48156222

Reply:
Indeed, this is still on CBO epic, e.g https://github.com/starburstdata/presto/commit/3765c8fe6ddd175f6631fcd694a92a294eec8716